x509/src/certificate.rs

```diff
@@ -33,160 +33,52 @@ impl Default for Version {
     }
 }
 
-/// X.509 `TBSCertificate` as defined in [RFC 5280 Section 4.1.2.5]
+/// X.509 `TBSCertificate` as defined in [RFC 5280 Section 4.1]
 ///
-/// ASN.1 structure containing the names of the subject and issuer, a public key associated
-/// with the subject, a validity period, and other associated information.
+/// ASN.1 structure containing the names of the subject and issuer, a public
+/// key associated with the subject, a validity period, and other associated
+/// information.
 ///
 /// ```text
-///   TBSCertificate  ::=  SEQUENCE  {
-///       version         [0]  Version DEFAULT v1,
-///       serialNumber         CertificateSerialNumber,
-///       signature            AlgorithmIdentifier{SIGNATURE-ALGORITHM, {SignatureAlgorithms}},
-///       issuer               Name,
-///       validity             Validity,
-///       subject              Name,
-///       subjectPublicKeyInfo SubjectPublicKeyInfo,
-///       ... ,
-///       [[2:               -- If present, version MUST be v2
-///       issuerUniqueID  [1]  IMPLICIT UniqueIdentifier OPTIONAL,
-///       subjectUniqueID [2]  IMPLICIT UniqueIdentifier OPTIONAL
-///       ]],
-///       [[3:               -- If present, version MUST be v3 --
-///       extensions      [3]  Extensions{{CertExtensions}} OPTIONAL
-///       ]], ... }
+/// TBSCertificate  ::=  SEQUENCE  {
+///     version         [0]  EXPLICIT Version DEFAULT v1,
+///     serialNumber         CertificateSerialNumber,
+///     signature            AlgorithmIdentifier,
+///     issuer               Name,
+///     validity             Validity,
+///     subject              Name,
+///     subjectPublicKeyInfo SubjectPublicKeyInfo,
+///     issuerUniqueID  [1]  IMPLICIT UniqueIdentifier OPTIONAL,
+///                          -- If present, version MUST be v2 or v3
+///     subjectUniqueID [2]  IMPLICIT UniqueIdentifier OPTIONAL,
+///                          -- If present, version MUST be v2 or v3
+///     extensions      [3]  Extensions OPTIONAL
+///                          -- If present, version MUST be v3 --
+/// }
 /// ```
 ///
-/// [RFC 5280 Section 4.1.2.5]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.5
-#[derive(Clone, Eq, PartialEq)]
-pub struct TBSCertificate<'a> {
-    /// version         [0]  Version DEFAULT v1,
-    //#[asn1(context_specific = "0", default = "Default::default")]
+/// [RFC 5280 Section 4.1.2.5]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.1
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+#[allow(missing_docs)]
+pub struct TbsCertificate<'a> {
```

Comment:
👍 on the name change. Was going to submit a PR for this myself

Reply:
It's odd clippy did not complain. The CRL and OCSP bits won't have this problem.

Reply:
Yeah, I'm a bit unclear on when `clippy` does or doesn't warn for these, but regardless this is the idiomatic capitalization in Rust.

Agree it would be good to retain the version comment. I'm actually kind of surprised it didn't trigger the `missing_docs` lint.